core/src/main/java/io/questdb/cairo/TableReader.java

```diff
@@ -63,11 +63,11 @@
     private final CairoConfiguration configuration;
     private final IntList symbolCountSnapshot = new IntList();
     private final TxReader txFile;
-    private final MappedReadOnlyMemory todoMem = new SinglePageMappedReadOnlyPageMemory();
+    private final MRMemory todoMem = new CMRMemoryImpl();
```

Comment:
maybe it will be good to have a factory to create implementations so that we can switch to another implementations

Reply:
```
class MemoryFactory {
    createCMRMemory()
    createCMARWMemory()
    createCARWMemory()
    createPARWMemory()
    createPMAMemory()
}
```

Reply:
done